bench/irmin-pack/tree.ml

```diff
@@ -197,39 +349,61 @@ struct
         let* tree = f tree in
         Store.Commit.v repo ~info:(info ()) ~parents:[ prev_commit ] tree
 
-  let add_commits repo ncommits f () =
+  let add_commits ~message repo ncommits f () =
+    with_progress_bar ~message ~n:ncommits ~unit:"commits" ~sampling_interval:1
+    @@ fun prog ->
     let* c = init_commit repo in
     let rec aux c i =
       if i >= ncommits then Lwt.return c
       else
         let* c' = checkout_and_commit repo (Store.Commit.hash c) f in
+        prog Int64.one;
         aux c' (i + 1)
     in
     let+ _ = aux c 0 in
     ()
 
-  let run ~mode config =
+  let run_large config =
     reset_stats ();
     let conf = Irmin_pack.config ~readonly:false ~fresh:true config.root in
     let* repo = Store.Repo.v conf in
     let* result =
-      (match mode with
-      | `Large ->
-          Trees.add_large_trees config.width config.nlarge_trees
-          |> add_commits repo config.ncommits
-      | `Chains ->
-          Trees.add_chain_trees config.depth config.nchain_trees
-          |> add_commits repo config.ncommits)
+      Trees.add_large_trees config.width config.nlarge_trees
+      |> add_commits ~message:"Playing large mode" repo config.ncommits
       |> Benchmark.run config
     in
     let+ () = Store.Repo.close repo in
-    (config, result)
+    fun fmt ->
```

Comment:
I see this is already in this file in a few places, but conventionally `fmt` is a format-string, and `ppf` is a Format.formatter.